contracts/strategies/SnowballStrategyV1.sol

```diff
@@ -0,0 +1,235 @@
+        updateAdminFee(_adminFeeBips);
+        updateDevFee(_devFeeBips);
+        updateReinvestReward(_reinvestRewardBips);
+        updateDepositsEnabled(true);
+        transferOwnership(_timelock);
+
+        emit Reinvest(0, 0);
+    }
+
+    /**
+     * @notice Initialization helper for Pair deposit tokens
+     * @dev Checks that selected Pairs are valid for trading reward tokens
+     * @dev Assigns values to swapPairToken0 and swapPairToken1
+     */
+    function assignSwapPairSafely(address _swapPairWAVAXSnob, address _swapPairToken0, address _swapPairToken1, address _rewardToken) private {
+        require(
+            DexLibrary.checkSwapPairCompatibility(IPair(_swapPairWAVAXSnob), address(WAVAX), address(_rewardToken)),
+            "_swapPairWAVAXSnob is not a WAVAX-Snob pair"
+        );
+        require(
+            _swapPairToken0 == address(0)
+            || DexLibrary.checkSwapPairCompatibility(IPair(_swapPairToken0), address(WAVAX), IPair(address(depositToken)).token0()),
+            "_swapPairToken0 is not a WAVAX+deposit token0"
+        );
+        require(
+            _swapPairToken1 == address(0)
+            || DexLibrary.checkSwapPairCompatibility(IPair(_swapPairToken1), address(WAVAX), IPair(address(depositToken)).token1()),
+            "_swapPairToken0 is not a WAVAX+deposit token1"
+        );
+        // converts Snob to WAVAX
+        swapPairWAVAXSnob = IPair(_swapPairWAVAXSnob);
+        // converts WAVAX to pair token0
+        swapPairToken0 = IPair(_swapPairToken0);
+        // converts WAVAX to pair token1
+        swapPairToken1 = IPair(_swapPairToken1);
+    }
+
+    function setAllowances() public override onlyOwner {
+        depositToken.approve(address(snowGlobe), MAX_UINT);
+        snowGlobe.approve(address(stakingContract), MAX_UINT);
+    }
+
+    function deposit(uint amount) external override {
+        _deposit(msg.sender, amount);
+    }
+
+    function depositWithPermit(uint amount, uint deadline, uint8 v, bytes32 r, bytes32 s) external override {
+        depositToken.permit(msg.sender, address(this), amount, deadline, v, r, s);
+        _deposit(msg.sender, amount);
+    }
+
+    function depositFor(address account, uint amount) external override {
+        _deposit(account, amount);
+    }
+
+    function _deposit(address account, uint amount) private onlyAllowedDeposits {
+        require(DEPOSITS_ENABLED == true, "SnowballStrategyV1::_deposit");
+        if (MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST > 0) {
+            uint unclaimedRewards = checkReward();
+            if (unclaimedRewards > MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST) {
+                _reinvest(unclaimedRewards);
+            }
+        }
+        require(depositToken.transferFrom(msg.sender, address(this), amount));
+        uint lpAmountDeposited = _stakeDepositTokens(amount);
+        _mint(account, getSharesForDepositTokens(lpAmountDeposited));
+        totalDeposits = totalDeposits.add(lpAmountDeposited);
```

Comment:
Is `lpAmountDeposited` used because it can be different than `amount`? Where does the undeposited balance go?

Reply:
Yes, snowball looses ~1-2 units when converting the LP to their shares, probably a rounding error. The loss is very small, I tested this a lot. If I don't calculate like they do 100% withdrawals won't work. Do we need an extra check to make sure the loss does not exceed expectation?